src/components/RestaurantList.tsx

```diff
@@ -1,65 +1,53 @@
 import { Category, Restaurant, SortingType } from '../types/restaurant';
 
-import React, { ChangeEvent } from 'react';
-import { RestaurantItem } from './';
+import React, { MouseEvent, ChangeEvent, useState } from 'react';
+import { Select, RestaurantItem } from './';
 import styled from 'styled-components';
 
+import filterRestaurants from '../domain/filterRestaurants';
 import { CATEGORIES, SORTING_TYPES } from '../constants';
 
 interface Props {
   restaurants: Restaurant[];
   openModal: (id: Restaurant['id']) => void;
-  setCategory: (category: Category) => void;
-  setSortingType: (sortingType: SortingType) => void;
 }
 
-class RestaurantList extends React.Component<Props> {
-  openDetailModal = (e: any) => {
-    const { id } = e.target.closest('li');
-    this.props.openModal(id);
-  };
+const RestaurantList = ({ restaurants, openModal }: Props) => {
+  const [category, setCategory] = useState<Category | '전체'>('전체');
+  const [sortingType, setSortingType] = useState<SortingType>('이름순');
+  const filteredRestaurants = filterRestaurants(restaurants, category, sortingType);
 
-  onChangeCategory = (e: ChangeEvent) => {
-    if (!(e.target instanceof HTMLSelectElement)) return;
+  const openDetailModal = (e: MouseEvent) => {
+    if (!(e.target instanceof HTMLElement)) return;
 
-    const { value: category } = e.target;
-    this.props.setCategory(category as Category);
+    const $li = e.target.closest('li');
+    if ($li) openModal($li.id as Restaurant['id']);
   };
 
-  onChangeSortingType = (e: ChangeEvent) => {
+  const onChangeSelect = (e: ChangeEvent) => {
     if (!(e.target instanceof HTMLSelectElement)) return;
-
-    const { value: sortingType } = e.target;
-    this.props.setSortingType(sortingType as SortingType);
+    const { name, value } = e.target;
+    if (name === 'category') setCategory(value as Category);
+    if (name === 'sorting') setSortingType(value as SortingType);
```

Comment:
의도는 선택상자마다 선택되는 이벤트 핸들러가 많아지면 코드가 길어져서 하나로 몰아보자는 의도로 작성하신 것 같아요.
요구사항이 여기서 더 추가될 리는 없어서 사실 지금의 수준으로도 무방하다고 보는데요!
다만, 선택 상자가 많아지는 요구사항을 받은 경우에도 이 방법이 유효할지도 추가적으로 고민을 해보시면 좋겠네요!

Reply:
`name`속성을 적극 활용하는 방법이 생각납니다!
select value 데이터를 커스텀 훅으로 만들어서 setState 수를 줄여도 좋을 것 같습니다.
```suggestion
    setSelect(name, value)
```